crates/router/src/core/payments/flows/psync_flow.rs

```diff
@@ -170,29 +170,27 @@ impl types::RouterData<api::PSync, types::PaymentsSyncData, types::PaymentsRespo
             )
             .await
             .to_payment_failed_response()?;
-            let capture_sync_response = match resp.response {
-                Err(err) => types::CaptureSyncResponse::Error {
-                    code: err.code,
-                    message: err.message,
-                    reason: err.reason,
-                    status_code: err.status_code,
+            match resp.response {
+                Err(err) => {
+                    capture_sync_response_map.insert(connector_capture_id, types::CaptureSyncResponse::Error {
+                        code: err.code,
+                        message: err.message,
+                        reason: err.reason,
+                        status_code: err.status_code,
+                    });
                 },
-                Ok(types::PaymentsResponseData::TransactionResponse {
-                    resource_id,
-                    connector_response_reference_id,
-                    ..
-                }) => types::CaptureSyncResponse::Success {
-                    resource_id,
-                    status: resp.status,
-                    connector_response_reference_id,
-                },
-                // this error is never meant to occur. response type will always be PaymentsResponseData::TransactionResponse
-                _ => Err(ApiErrorResponse::PreconditionFailed { message: "Response type must be PaymentsResponseData::TransactionResponse for payment sync".into() })?,
+                Ok(types::PaymentsResponseData::MultipleCaptureResponse { capture_sync_response_list })=> {
+                    capture_sync_response_map.extend(capture_sync_response_list.into_iter());
+                }
+                _ => Err(ApiErrorResponse::PreconditionFailed { message: "Response type must be PaymentsResponseData::MultipleCaptureResponse for payment sync".into() })?,
             };
-            capture_sync_response_list.insert(connector_capture_id.clone(), capture_sync_response);
+            if let payments::CallConnectorAction::HandleResponse(_) = call_connector_action {
```

Comment:
Let's keep this logic outside for loop, Incase of HandleResponse execute_connector_processing_step should be called only once and For trigger use the existing loop